src/sth_database.js

```diff
@@ -169,52 +171,79 @@
    * @return {string} The database name
    */
   function getDatabase(service) {
-    return sthConfig.DB_PREFIX + '_' + service;
+    return sthConfig.DB_PREFIX + service;
   }
 
   /**
    * Return the name of the collection which will store the raw events
+   * @param {string} databaseName The database name
    * @param {string} servicePath The service path of the entity the event is related to
    * @param {string} entityId The entity id related to the event
    * @param {string} entityType The type of entity related to the event
    * @param {string} attrName The attribute id related to the event
    * @returns {string} The collection name
    */
-  function getCollectionName4Events(servicePath, entityId, entityType, attrName) {
+  function getCollectionName4Events(databaseName, servicePath, entityId, entityType, attrName) {
+    var collectionName4Events;
     switch(sthConfig.DATA_MODEL) {
       case sthConfig.DATA_MODELS.COLLECTIONS_PER_SERVICE_PATH:
-        return sthConfig.COLLECTION_PREFIX + '_' + servicePath;
+        collectionName4Events = servicePath;
+        break;
       case sthConfig.DATA_MODELS.COLLECTIONS_PER_ENTITY:
-        return sthConfig.COLLECTION_PREFIX + '_' + servicePath + '_' + entityId + (entityType ? '_' + entityType : '');
+        collectionName4Events =  servicePath + '_' + entityId + (entityType ? '_' + entityType : '');
+        break;
       case sthConfig.DATA_MODELS.COLLECTIONS_PER_ATTRIBUTE:
-        return sthConfig.COLLECTION_PREFIX + '_' + servicePath + '_' + entityId + (entityType ? '_' + entityType : '') +
+        collectionName4Events =  servicePath + '_' + entityId + (entityType ? '_' + entityType : '') +
           '_' + attrName;
+        break;
     }
+    // The maximum number of bytes accepted by MongoDB for namespaces is 120 bytes.
+    var limit = 119 - bytesCounter.count(databaseName) - bytesCounter.count(sthConfig.COLLECTION_PREFIX) -
+      bytesCounter.count('.aggr');
+    return sthConfig.COLLECTION_PREFIX + generateHash(collectionName4Events, limit);
```

Comment:
That's right! In fact, the number of bytes of a character enconded in UTF-8 can be up to 4 bytes (mostly for Eastern languages :O ) 

On the other hand, take into consideration that currently the `sthConfig.COLLECTION_PREFIX` is also free text so it could be any length for the time being :O

On the other hand, 10 bytes allows many combinations :D and the collision probability may not be that low :) In fact, for 10 bytes we are talking about 2^80 combinations, this is 1,20*10^24 :O
To avoid collisions we could include a variable/increasing part in the seed to calculate the hash. Currently I use the concatenation of service path, entityId, entityType and attributeName (depending on the model of course), but we could add for example the number of milliseconds since 1970 to this concatenation, just as MongoDB does to automatically generate the _id for docs :) This technique would also allow us to recalculate the hash in case of a collision since this last value will change and a new hash will be generated :)

I do not get the part about enforcing 1 byte encoding... :) Do you mean forcing the user to use only characters encoded in UTF-8 as 1 byte (not special characters such as ñ, á, é, í, etc.)?

Thoughts are more than welcome :)